common/src/types.rs

```diff
@@ -49,3 +65,27 @@ impl<'de> Deserialize<'de> for BlockTag {
         Ok(block_tag)
     }
 }
+
+impl From<BlockTag> for BlockId {
```

Comment:
Why do we have both `BlockTag` and `BlockId`? Can we use just one of them throughout the codebase?

Reply:
`BlockTag` is a custom type used for the block param in Helios execution RPC interface. It is restricted to `latest`/`finalized`/`<number>` to facilitate fetching a block from Helios state.

Whereas `BlockId` comes from Alloy and has values like `<hash>`/`pending`/`safe`/`earliest` (in addition to the ones above) and is used by alloy's RPC provider interface.

To use only `BlockId` throughout the codebase, we would need to handle the `pending`/`safe`/`earliest` cases somehow.

Could create a separate issue for this, might be out of scope for this PR since `BlockTag` already exists and is not a new addition in this PR.

Reply:
Yeah we can leave this be for now.